Islandora/islandora
# captions v2

GitHub Issue:
Islandora/documentation#1305
Islandora/documentation#1003
https://www.drupal.org/project/drupal/issues/3057144
https://www.drupal.org/project/drupal/issues/3056714

Other Relevant Links
Perhaps superseding https://github.com/Islandora/islandora/pull/743/files
Definitely supersedes https://github.com/Islandora/islandora_defaults/pull/44

# What does this Pull Request do?
Takes the work in the patch 16 here https://www.drupal.org/project/drupal/issues/3056714  
Allows a user to add a "track" to an audio or video media object which will play in the HTML5 player

# What's new?
* code related to the config in https://github.com/Islandora/islandora_defaults/pull/48
* new templates, fieldtype, fieldwidget, fieldformatters

# How should this be tested?
* spin up an islandora site
* pull in islandora_defaults PR https://github.com/Islandora/islandora_defaults/pull/48 and this islandora PR
* do a partial config import on the config/install directory of islandora_defaults
* cache clear
* add an audio and/or video node
* add an audio and/or video original file media attached to that node
* when the service file derivative is created, add a VTT caption file to that service file media in the new track field
* view the node - in the menu on the HTML5 player, you should be able to toggle on and off captions

# Additional Notes
* this does not have the poster/thumbnail feature i presented in https://github.com/Islandora/islandora_defaults/pull/44

# Interested parties
@Islandora/8-x-committers @Natkeeran
